paasta_tools/mesos_tools.py

```diff
@@ -226,13 +245,69 @@ def format_non_running_mesos_task_row(task, get_short_task_id):
     )
 
 
-def status_mesos_tasks_verbose(job_id, get_short_task_id):
+@timeout()
+def format_stdstreams_tail_for_task(task, get_short_task_id, nlines=10):
+    """Returns the formatted "tail" of stdout/stderr, for a given a task.
+
+    :param get_short_task_id: A function which given a
+                              task_id returns a short task_id suitable for
+                              printing.
+    """
+    error_message = PaastaColors.red("      couldn't read stdout/stderr for %s (%s)")
+    output = []
+    try:
+        fobjs = list(mesos.cli.cluster.files(flist=['stdout', 'stderr'], fltr=task['id']))
+        fobjs.sort(key=lambda fobj: fobj[0].path, reverse=True)
+        if not fobjs:
+            output.append(PaastaColors.blue("      no stdout/stderrr for %s" % get_short_task_id(task['id'])))
+            return output
+        for fobj in fobjs:
+            output.append(PaastaColors.blue("      %s tail for %s" % (fobj[0].path, get_short_task_id(task['id']))))
+            # read nlines, starting from EOF
+            # mesos.cli is smart and can efficiently read a file backwards
+            reversed_file = reversed(fobj[0])
+            tail = []
+            for _ in xrange(nlines):
+                line = next(reversed_file, None)
+                if line is None:
+                    break
+                tail.append(line)
+            # reverse the tail, so that EOF is at the bottom again
+            if tail:
+                output.extend(tail[::-1])
+            output.append(PaastaColors.blue("      %s EOF" % fobj[0].path))
+    except (MasterNotAvailableException,
```

Comment:
can you be more specific about where these exceptions are thrown? as in, don't wrap this whole code block in try/catch - just the parts that will actually throw